compiler/rustc_lint/src/builtin.rs

```diff
@@ -547,7 +547,7 @@ impl<'tcx> LateLintPass<'tcx> for MissingDoc {
 
     fn check_item(&mut self, cx: &LateContext<'_>, it: &hir::Item<'_>) {
         match it.kind {
-            hir::ItemKind::Trait(..) => {
+            hir::ItemKind::Trait(..) | hir::ItemKind::TraitAlias(..) => {
                 // Issue #11592: traits are always considered exported, even when private.
```

Comment:
This comment is very outdated.
The special case for traits and trait aliases can likely be removed, but it may require some more understanding of the surrounding logic.

Reply:
> This comment is very outdated. The special case for traits and trait aliases can likely be removed, but it may require some more understanding of the surrounding logic.

Getting rid of the check, tests/ui/issues/issue-11592.rs still passes, is there anything else to look out for there?

Reply:
>Getting rid of the check, tests/ui/issues/issue-11592.rs still passes, is there anything else to look out for there?

If test cases from https://github.com/rust-lang/rust/issues/111025 fail before the change, pass after the change, and nothing else breaks, than just removing the check is probably enough.